iroh/examples/echo-no-router.rs

```diff
@@ -0,0 +1,102 @@
+//! Very basic example showing how to implement a basic echo protocol,
+//! without using the `Router` API. (For the router version, check out the echo.rs example.)
+//!
+//! The echo protocol echos any data sent to it in the first stream.
+//!
+//! ## Running the Example
+//!
+//!     cargo run --example echo-no-router --features=examples
+
+use anyhow::Result;
+use iroh::{Endpoint, NodeAddr};
+
+/// Each protocol is identified by its ALPN string.
+///
+/// The ALPN, or application-layer protocol negotiation, is exchanged in the connection handshake,
+/// and the connection is aborted unless both nodes pass the same bytestring.
+const ALPN: &[u8] = b"iroh-example/echo/0";
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let endpoint = accept_side().await?;
+    let node_addr = endpoint.node_addr().await?;
+
+    connect_side(node_addr).await?;
+
+    endpoint.close().await;
+
+    Ok(())
+}
+
+async fn connect_side(addr: NodeAddr) -> Result<()> {
+    let endpoint = Endpoint::builder().discovery_n0().bind().await?;
+
+    // Open a connection to the accepting node
+    let conn = endpoint.connect(addr, ALPN).await?;
+
+    // Open a bidirectional QUIC stream
+    let (mut send, mut recv) = conn.open_bi().await?;
+
+    // Send some data to be echoed
+    send.write_all(b"Hello, world!").await?;
+
+    // Signal the end of transfer on this particular stream.
+    send.finish()?;
+
+    // Receive the echo
+    let response = recv.read_to_end(1000).await?;
+    assert_eq!(&response, b"Hello, world!");
+
+    // Explicitly close the whole connection.
+    conn.close(0u32.into(), b"bye!");
+
+    Ok(())
+}
+
+async fn accept_side() -> Result<Endpoint> {
+    let endpoint = Endpoint::builder()
+        .discovery_n0()
+        .alpns(vec![ALPN.to_vec()])
+        .bind()
+        .await?;
+
+    let ep = endpoint.clone(); // Cloning so we can both return it and use it in the tokio task
+
+    // spawn a task so that `accept_side` returns immediately and we can continue in main().
+    tokio::spawn(async move {
```

Comment:
maybe make a mention that you should track all these tasks in a more serious environement